api/src/services/graphql/utils/process-error.ts

```diff
@@ -1,12 +1,19 @@
-import { isDirectusError } from '@directus/errors';
+import { isDirectusError, type DirectusError } from '@directus/errors';
 import type { Accountability } from '@directus/types';
 import type { GraphQLError, GraphQLFormattedError } from 'graphql';
 import logger from '../../../logger.js';
 
-const processError = (accountability: Accountability | null, error: Readonly<GraphQLError>): GraphQLFormattedError => {
+const processError = (
+	accountability: Accountability | null,
+	error: Readonly<GraphQLError & { originalError: GraphQLError | DirectusError | Error | undefined }>,
+): GraphQLFormattedError => {
 	logger.error(error);
 
-	const { originalError } = error;
+	let originalError = error.originalError;
+
+	if (originalError && 'originalError' in originalError) {
+		originalError = originalError.originalError;
+	}
```

Comment:
even though `originalError` is always nested inside `error.originalError`
I have added this logic to make sure to keep it backward compatible or inline with the `GraphQLError` type definition.